aten/src/ATen/native/ForeachUtils.h

```diff
@@ -207,22 +207,38 @@ FlatMap _group_tensors_by_first_tensors_device_and_dtype(const nested_optional_t
         TORCH_CHECK(
           t.has_value(),
           "Tensors of the first list of nested Tensor lists are supposed to be defined but ",
-          tensor_index, "the Tensor is not.");
+          tensor_index, "-th Tensor is not.");
```

Comment:
```suggestion
          "the ", tensor_index, "-th Tensor is not.");
```
also nit

Thank you for fixing my wrong fix from earlier 🥲